emap/emap.go

```diff
@@ -48,14 +48,16 @@ func (e *EMap[T]) Add(items []T) {
 	defer e.mu.Unlock()
 
 	for _, item := range items {
-		e.add(item.ID(), item.Expiry())
+		e.add(item.ID(), reducePrecision(item.Expiry()))
```

Comment:
If we use this approach, we may need whoever calls into emap to ensure they don't re-add below this reduced precision boundary. May make sense for the wrapper to handle this instead?